routing/blindedpath/blinded_path.go

```diff
@@ -435,11 +437,27 @@ func collectRelayInfo(cfg *BuildBlindedPathCfg, path *candidatePath) (
 			}
 		}
 
-		policy, err = cfg.AddPolicyBuffer(policy)
+		if policy.MinHTLCMsat > cfg.ValueMsat {
+			return nil, 0, 0, fmt.Errorf("minHTLC of hop policy "+
```

Comment:
I think this should wrap an `errInvalidBlindedPath`, like how we handle `if minHTLC > maxHTLC {` below.